### DISCUSSION TITLE
Override /livewire/upload-file and /livewire/preview-file/{filename} routes

### DISCUSSION THREAD

**Discussion by EnriqueSanVic** [livewire/livewire#7480] [3. Feature ideas]
I use a Laravel instance as a subpath on my main domain, my main server is made with another technology and the Laravel root starts from `mydomain.com/laravelservice`.

When using Livewire the routes that are automatically declared start in the root path, for example `mydomain.com/livewire/livewire.js` or `mydomain.com/livewire/update`.

To override the routes that Livewire automatically declares and set the prefix of my Laravel base path, i use the route override static methods `Livewire::setScriptRoute` and `Livewire::setUpdateRoute` used on the `boot` method in my `AppServiceProvider`.  After implementing this, the Livewire paths change to `mydomain.com/aravelservice/livewire/livewire.js`  and `mydomain.com/aravelservice/livewire/update`. 

```php

$prefixPath = "laravelservice";

Livewire::setScriptRoute(function ($handle) use ($assetUrl) {
    return Route::get($prefixPath . '/livewire/livewire.js', $handle);
});
Livewire::setUpdateRoute(function ($handle) use ($assetUrl) {
    return Route::post($prefixPath . '/livewire/update', $handle)->name('laravelservice.livewire.update');
});
```

This works correctly, the problem is when I want to use livewire's file upload support features. For this purpose Livewire automatically declares the paths `mydomain.com/livewire/upload-file` and `mydomain.com/livewire/preview-file/{filename}` but these routes cannot be overridden by static methods such as `Livewire::setScriptRoute` and `Livewire::setUpdateRoute`.

I've been looking at the Livewire code and found the declaration of these routes in `Features\SupportFileUploads\SupportFileUploads.php:34`:

```php
Route::post('/livewire/upload-file', [FileUploadController::class, 'handle'])
  ->name('livewire.upload-file')
  ->middleware('web');

Route::get('/livewire/preview-file/{filename}', [FilePreviewController::class, 'handle'])
  ->name('livewire.preview-file')
  ->middleware('web');
```

Apparently the declaration of these routes is not parameterized. In my case I need to be able to add a prefix to these routes and I would like to be able to override them like the routes `/livewire/livewire.js` and `/livewire/update` and the static methods `Livewire::setScriptRoute` and `Livewire::setUpdateRoute`.

**Comment by joshhanley** [comment]
@EnriqueSanVic there was a PR for this PR #6803 but it was incomplete. So we're open to the idea if someone wants to submit a complete PR 🙂
